bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/device/CommandHandler.java

```diff
@@ -114,22 +112,24 @@ protected String nm() {
 
     protected int getMaxLightLevel(InsteonChannelConfiguration conf, int defaultLevel) {
         Map<String, @Nullable String> params = conf.getParameters();
-        if (conf.getFeature().contains("dimmer") && params.containsKey("dimmermax")) {
-            String item = conf.getChannelName();
+        if (conf.getFeature().contains("dimmer")) {
             String dimmerMax = params.get("dimmermax");
-            try {
-                int i = Integer.parseInt(dimmerMax);
-                if (i > 1 && i <= 99) {
-                    int level = (int) Math.ceil((i * 255.0) / 100); // round up
-                    if (level < defaultLevel) {
-                        logger.debug("item {}: using dimmermax value of {}", item, dimmerMax);
-                        return level;
+            if (dimmerMax != null) {
```

Comment:
Perhaps this if condition can be added to `if (conf.getFeature().contains("dimmer")) {` to simplify the code?

Reply:
Not really, getFeature() returns a string and the first if check to see it it contains the string "dimmer". If it does the second if checks to see if there is a parameter called "dimmermax".